src/datasources/email-api/pushwoosh-api.service.ts

```diff
@@ -63,4 +63,18 @@ export class PushwooshApi implements IEmailApi {
       throw this.httpErrorFactory.from(error);
     }
   }
+
+  async deleteEmailAddress(args: { emailAddress: string }): Promise<void> {
+    try {
+      const url = `${this.baseUri}/json/1.3/deleteEmail`;
+      await this.networkService.post(url, {
+        request: {
+          application: this.applicationCode,
```

Comment:
The API key is not required for this call?

Reply:
Apparently, it isn't: https://docs.pushwoosh.com/platform-docs/api-reference/email-api#deleteemail

I tried this without authentication and it seems to work (returns a 200 status code). Email addresses are registered automatically to the _PW application_ when creating an email message, but it seems (un)registration endpoints are permissionless. I guess it's intended to be used as a way to (un)subscribe users on their own.